imports/lib/models/Model.ts

```diff
@@ -525,6 +530,20 @@ class Model<Schema extends MongoRecordZodType, IdSchema extends z.ZodTypeAny = t
   }
 }
 
-export type ModelType<M extends Model<any, any>> = z.output<M['schema']>;
+export type ModelType<M extends Model<any, any, any>> = z.output<M['schema']>;
+
+export type ForeignKeyType<M extends Model<any, any, any>> = string & z.BRAND<M['name']>;
+export const makeForeignKey = <M extends Model<any, any, any>>(id: string): ForeignKeyType<M> => {
+  return id as ForeignKeyType<M>;
+};
+
+export const matchForeignKey = <
```

Comment:
This is a function that returns a matcher function, not the matcher function itself, so I'd call this makeForeignKeyMatcher<M>() or something so it's more apparent that you're supposed to invoke it in your `match()` call when you wouldn't do the same for e.g. `String`